libs/market-depth/src/lib/orderbook-manager.tsx

```diff
@@ -1,113 +1,89 @@
 import { marketDataProvider, marketProvider } from '@vegaprotocol/markets';
-import type { MarketData } from '@vegaprotocol/markets';
-import { useCallback, useEffect, useRef, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import type {
-  MarketDepthUpdateSubscription,
   MarketDepthQuery,
   MarketDepthQueryVariables,
+  MarketDepthUpdateSubscription,
+  PriceLevelFieldsFragment,
 } from './__generated__/MarketDepth';
-import type { PriceLevelFieldsFragment } from './__generated__/MarketDepth';
-import {
-  compactRows,
-  updateCompactedRows,
-  getMidPrice,
-  getPriceLevel,
-} from './orderbook-data';
-import type { OrderbookData } from './orderbook-data';
+import { VolumeType, fillSubscriptionData } from './orderbook-data';
 import { useOrderStore } from '@vegaprotocol/orders';
 
+export type OrderbookData = {
+  asks: PriceLevelFieldsFragment[];
+  bids: PriceLevelFieldsFragment[];
+};
+
 interface OrderbookManagerProps {
   marketId: string;
 }
 
 export const OrderbookManager = ({ marketId }: OrderbookManagerProps) => {
-  const [resolution, setResolution] = useState(1);
   const variables = { marketId };
-  const resolutionRef = useRef(resolution);
-  const [orderbookData, setOrderbookData] = useState<OrderbookData>({
-    rows: null,
-  });
-  const dataRef = useRef<OrderbookData>({ rows: null });
-  const marketDataRef = useRef<MarketData | null>(null);
-  const rawDataRef = useRef<MarketDepthQuery['market'] | null>(null);
-  const deltaRef = useRef<{
-    sell: PriceLevelFieldsFragment[];
-    buy: PriceLevelFieldsFragment[];
-  }>({
-    sell: [],
-    buy: [],
+  const [orderbookData, setOrderBookData] = useState<OrderbookData>({
+    asks: [],
+    bids: [],
   });
-  const updateOrderbookData = useRef(
-    throttle(() => {
-      dataRef.current = {
-        ...marketDataRef.current,
-        indicativePrice:
-          marketDataRef.current?.indicativePrice &&
-          getPriceLevel(
-            marketDataRef.current.indicativePrice,
-            resolutionRef.current
-          ),
-        midPrice: getMidPrice(
-          rawDataRef.current?.depth.sell,
-          rawDataRef.current?.depth.buy,
-          resolution
-        ),
-        rows:
-          deltaRef.current.buy.length || deltaRef.current.sell.length
-            ? updateCompactedRows(
-                dataRef.current.rows ?? [],
-                deltaRef.current.sell,
-                deltaRef.current.buy,
-                resolutionRef.current
-              )
-            : dataRef.current.rows,
-      };
-      deltaRef.current.buy = [];
-      deltaRef.current.sell = [];
-      setOrderbookData(dataRef.current);
-    }, 250)
-  );
 
-  useEffect(() => {
-    deltaRef.current.buy = [];
-    deltaRef.current.sell = [];
-  }, [marketId]);
+  const initData = useCallback((data?: MarketDepthQuery['market']) => {
+    const bids = (data?.depth.buy || []).filter((item) => item.volume !== '0');
```

Comment:
If there is a need to filter out zero volume price levels it would be better to do it in data provider